### PR TITLE
Fixes for plugins:

### PR DESCRIPTION
**Pull request by jdrossl**
- Move load file endpoint to webapp
- Fix base url in the configuration
- Add path validations

https://github.com/craftercms/craftercms/issues/4565

### CODE REVIEW

File: src/main/java/org/craftercms/studio/controller/web/PluginController.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+import javax.activation.MimetypesFileTypeMap;
+
+/**
+ * Controller that provides the UI with plugin related files
+ * @author joseross
+ * @since 4.0
+ */
+@RestController
+@RequestMapping("/plugin")
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/controller/web/PluginController.java]: Should this be /plugin/1/{API}?

File: src/main/java/org/craftercms/studio/controller/web/PluginController.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    protected final ConfigurationService configurationService;
+
+    public PluginController(ConfigurationService configurationService) {
+        this.configurationService = configurationService;
+    }
+
+    /**
+     * Returns a single file for a given plugin
+     */
+    @GetMapping("/file")
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/controller/web/PluginController.java]: or the /1/ goes here so it's per API?

File: src/main/resources/crafter/studio/studio-config.yaml [hunk 3]
```diff
@@ -204,7 +204,7 @@ studio.configuration.management.previewProtectedUrls: >-
   /api/1/site/cache/statistics
 
 # The pattern used to load plugins installed from the marketplaces
-studio.configuration.plugin.base.pattern: /config/studio/plugins/${id}/${type}/${name}
+studio.configuration.plugin.base.pattern: /config/studio/plugins/js/${id}/${type}/${name}
```
Comment (sumerjabri) [inline src/main/resources/crafter/studio/studio-config.yaml:207]: Is this a fix to a prior bug?

**Review by sumerjabri — CHANGES_REQUESTED** [review]

File: src/main/java/org/craftercms/studio/controller/web/PluginController.java (hunk 1 above)
Comment (jdrossl) [reply src/main/java/org/craftercms/studio/controller/web/PluginController.java]: I would suggest `/1/plugin`

File: src/main/java/org/craftercms/studio/controller/web/PluginController.java (hunk 2 above)
Comment (jdrossl) [reply src/main/java/org/craftercms/studio/controller/web/PluginController.java]: see above

File: src/main/resources/crafter/studio/studio-config.yaml (hunk 3 above)
Comment (jdrossl) [reply src/main/resources/crafter/studio/studio-config.yaml:207]: Yes, it was introduced with the PR for installing plugins, it's new code that none has used yet

File: src/main/java/org/craftercms/studio/controller/web/PluginController.java (hunk 1 above)
Comment (sumerjabri) [reply src/main/java/org/craftercms/studio/controller/web/PluginController.java]: Let's do it.

**Review by sumerjabri — APPROVED** [review]

**Pull request merged**